### ISSUE TITLE
Inventory Generators crash on any custom fuels.

### ISSUE THREAD

**Issue by ch0xy** [AllTheMods/ATM-4#109]
I put the Dwarf coal inside of my extreme inventory generator and it instantly crashed my game on stream and corrupted the world with a ticking player.

**Comment by LezChap** [comment]
1) Crashlogs please.
2) Probably an issue with either the Dwarf Coal or Generators mods...hope you're reporting it to them (with the crash logs).

Probably the easiest fix to get back into your world is to use NBTExplorer or other NBT Editor to delete the generator item from your inventory.

**Comment by ch0xy** [comment]
[crash-2020-01-02_17.25.31-server.txt](https://github.com/AllTheMods/ATM-4/files/4020106/crash-2020-01-02_17.25.31-server.txt)

**Comment by LezChap** [comment]
> Probably the easiest fix to get back into your world is to use NBTExplorer or other NBT Editor to delete the generator item from your inventory.

**Comment by ch0xy** [comment]
> > Probably the easiest fix to get back into your world is to use NBTExplorer or other NBT Editor to delete the generator item from your inventory.

Yeah I just did that I am going to test and try to load world again.

**Comment by ch0xy** [comment]
[crash-2020-01-03_09.34.31-server.txt](https://github.com/AllTheMods/ATM-4/files/4020190/crash-2020-01-03_09.34.31-server.txt)
This is the current crash log after the generator is removed from inventory.

**Comment by ch0xy** [comment]
The only way I could get it fixed is by removing the mod file. Then run the game have it delete everything in the world then restart and put mod back in.

**Comment by ch0xy** [comment]
So I just found out last night even the firestone stuff to make torches as fule inside of it caused crashes so it is not related to just dwarf coal now.  I have removed the inventory generators from my pack after it has corrupted my world 5 times.

**Comment by rswester** [comment]
Just had the same problem aswell with a Diesel Canister.
Mostlikely it's a bug in the "inventory generators" mod.

**Comment by Bayarlakh** [comment]
If the thing in the fuel slot of the inventory generator is ANYTHING(even accidentally shift clicking upgrades for the generator from "Inventory Generators" mod puts it in the fuel slot and crashes) besides vanilla(as far as i know planks, logs and coal) fuel it just crashes and now my world is dead.
